### PR TITLE
[YUNIKORN-2598] Update the unit test for CheckQueues and checkNodeSortingPolicy functions

### PR DESCRIPTION
**Pull request by SophieTech88**
### What is this PR for?
There are many methods not covered by configvalidator_test.go. This commit works on unit Test for checkQueues function.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* Update the unit Test for checkQueues function.

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2598

### How should this be tested?
This is a unite test. All existing unit tests should pass.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/common/configs/configvalidator_test.go [hunk 1]
```diff
@@ -2012,3 +2012,227 @@ func TestCheckQueuesStructure(t *testing.T) {
...
+				Limits: []Limit{
+					{
+						Limit: "user-limit",
+						Users: []string{"user1", "user2", "user1"},
+					},
+				},
+			},
+			level:            0,
+			errorExpected:    true,
+			expectedErrorMsg: "duplicated user name user1 , already existed",
```
Comment (pbacsko) [inline pkg/common/configs/configvalidator_test.go:2106]: This error message needs to be fixed in the config validator. 
It should be "duplicate user name user1, already exists". Or just simply `user name "user1" already exists`. There's no need for the extra whitespaces either.

File: pkg/common/configs/configvalidator_test.go [hunk 2]
```diff
@@ -2012,3 +2012,227 @@ func TestCheckQueuesStructure(t *testing.T) {
...
+				AdminACL:  "admin",
+				SubmitACL: "submit",
+				Queues: []QueueConfig{
+					{Name: "duplicateQueue"},
+					{Name: "duplicateQueue"},
+				},
+			},
+			level:            0,
+			errorExpected:    true,
+			expectedErrorMsg: "duplicate child name found with name duplicateQueue, level 0",
```
Comment (pbacsko) [inline pkg/common/configs/configvalidator_test.go:2062]: Let's put duplicateQueue in single quotes for consistency: `duplicate child name found with name 'duplicateQueue', level 0`

File: pkg/common/configs/configvalidator_test.go [hunk 3]
```diff
@@ -2012,3 +2012,227 @@ func TestCheckQueuesStructure(t *testing.T) {
...
+						Queues: []QueueConfig{
+							{Name: "duplicateQueue"},
+							{Name: "duplicateQueue"},
+						},
+					},
+				},
+			},
+			level:            0,
+			errorExpected:    true,
+			expectedErrorMsg: "duplicate child name found with name duplicateQueue, level 1",
```
Comment (pbacsko) [inline pkg/common/configs/configvalidator_test.go:2084]: same here

**Review by pbacsko — CHANGES_REQUESTED** [review]
Minor comments, they are related to the existing error messages. The PR itself looks good but please fix the error messages in the config validator.

Also add `//nolint:funlen` to the test function.

File: pkg/common/configs/configvalidator_test.go (hunk 1 above)
Comment (SophieTech88) [reply pkg/common/configs/configvalidator_test.go:2106]: for checkLimit function, it check the user name is duplicate or not. 
The original code is [here](https://github.com/apache/yunikorn-core/blob/master/pkg/common/configs/configvalidator.go#L491):
```
if existedUserName[name] {
    return fmt.Errorf("duplicated user name %s , already existed", name)
}
```
And I just want to trigger the error of `checkLimits` for `chckQueues` function. 
So the expectedErrorMsg is `duplicated user name user1 , already existed`. The whitespaces exist . So I need to match the expectedErrorMsg to the return error.
Do I need to change the both error message?

File: pkg/common/configs/configvalidator_test.go (hunk 2 above)
Comment (SophieTech88) [reply pkg/common/configs/configvalidator_test.go:2062]: Something similar here, the expectedErrorMsg need to match the return error when it call the function `checkQueues`, the error message to duplicate name is [here](https://github.com/apache/yunikorn-core/blob/master/pkg/common/configs/configvalidator.go#L660):
```
if queueMap[strings.ToLower(child.Name)] {
    return fmt.Errorf("duplicate child name found with name %s, level %d", child.Name, level)
}
```
Do I need to add single quote `'` to the error messages for both?

File: pkg/common/configs/configvalidator_test.go (hunk 1 above)
Comment (pbacsko) [reply pkg/common/configs/configvalidator_test.go:2106]: Yes, you need to change both, that was my request. The error message is weird, so why not fix it?

Comment (SophieTech88) [reply pkg/common/configs/configvalidator_test.go:2106]: Sure. Let me update both. Thanks.

File: pkg/common/configs/configvalidator_test.go [hunk 4]
```diff
@@ -2012,3 +2012,227 @@ func TestCheckQueuesStructure(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckQueues(t *testing.T) { //nolint:funlen
+	testCases := []struct {
+		name             string
+		queue            *QueueConfig
+		level            int
+		errorExpected    bool
```
Comment (chia7712) [inline pkg/common/configs/configvalidator_test.go:2021]: It seems `validateFunc` is not called if `errorExpected` is true. Maybe we can merge them to simplify code.

**Review by chia7712 — COMMENTED** [review]
@SophieTech88 thanks for your patch. one small comment left.

File: pkg/common/configs/configvalidator_test.go (hunk 4 above)
Comment (SophieTech88) [reply pkg/common/configs/configvalidator_test.go]: Update the code to remove the errorExpected variable and simplify code. Many thanks for the suggestion.

File: pkg/common/configs/configvalidator_test.go (hunk 2 above)
Comment (SophieTech88) [reply pkg/common/configs/configvalidator_test.go]: Update the code.

File: pkg/common/configs/configvalidator.go [hunk 5]
```diff
@@ -489,7 +489,7 @@ func checkLimit(limit Limit, existedUserName map[string]bool, existedGroupName m
 		}
 
 		if existedUserName[name] {
```
Comment (pbacsko) [inline pkg/common/configs/configvalidator.go:491]: We need some spell check here as well... let's do it in a different JIRA.

Comment (SophieTech88) [reply pkg/common/configs/configvalidator.go:491]: Sure.

**Comment by SophieTech88** [comment]
@pbacsko Many thanks for the comments and suggestions. Just make a follow-up to check whether there is any thing I need to do for this PR. Thanks

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@SophieTech88 thanks for your contribution!

**Pull request closed without merging**